GPU/GPUTracking/Merger/GPUTPCGMMerger.cxx

```diff
@@ -1919,3 +1935,61 @@ GPUd() void GPUTPCGMMerger::Finalize2(int nBlocks, int nThreads, int iBlock, int
     }
   }
 }
+
+struct MergeLooperParam {
+  float absz;
+  float tgl;
+  float qpt;
+  float x;
+  float y;
+  unsigned int id;
+};
+
+GPUd() void GPUTPCGMMerger::MergeLoopers(int nBlocks, int nThreads, int iBlock, int iThread)
+{
+  if (iThread || iBlock) {
+    return;
+  }
+#ifndef GPUCA_GPUCODE
+  std::vector<MergeLooperParam> params;
+  for (unsigned int i = 0; i < mMemory->nOutputTracks; i++) {
+    const auto& p = mOutputTracks[i].GetParam();
+    if (mOutputTracks[i].NClusters() && CAMath::Abs(p.GetQPt()) > 5.f) {
+      const int slice = mClusters[mOutputTracks[i].FirstClusterRef() + mOutputTracks[i].NClusters() - 1].slice;
+      const float z = p.GetZ() + (Param().earlyTpcTransform ? p.GetTZOffset() : GetConstantMem()->calibObjects.fastTransform->convVertexTimeToZOffset(slice, p.GetTZOffset(), Param().continuousMaxTimeBin));
+      float sinA, cosA;
+      CAMath::SinCos(Param().Alpha(slice), sinA, cosA);
+      float gx = cosA * p.GetX() + sinA * p.GetY();
+      float gy = cosA * p.GetY() - sinA * p.GetX();
+      float bz = Param().polynomialField.GetFieldBz(gx, gy, z);
+      const float r1 = CAMath::Abs(p.GetQPt() * bz);
+      const float r = r1 > 0.0001 ? (1.f / CAMath::Abs(r1)) : 10000;
+      const float mx = p.GetX() + r * p.GetSinPhi();
+      const float my = p.GetY() + r * CAMath::Sqrt(1 - p.GetSinPhi() * p.GetSinPhi());
+      const float gmx = cosA * mx + sinA * my;
+      const float gmy = cosA * my - sinA * mx;
+      params.emplace_back(MergeLooperParam{CAMath::Abs(z), CAMath::Abs(p.GetDzDs()), p.GetDzDs() > 0 ? p.GetQPt() : -p.GetQPt(), gmx, gmy, i});
+    }
+  }
+  std::sort(params.begin(), params.end(), [](const MergeLooperParam& a, const MergeLooperParam& b) { return a.absz < b.absz; });
+  for (unsigned int i = 0; i < params.size(); i++) {
+    for (unsigned int j = i + 1; j < params.size(); j++) {
+      if (params[j].absz > params[i].absz + 100.f) {
+        break;
+      }
+      bool EQ = CAMath::Abs(params[i].x - params[j].x) < 10.f && CAMath::Abs(params[i].y - params[j].y) < 10.f && CAMath::Abs(params[i].tgl - params[j].tgl) < 0.15f && CAMath::Abs((params[i].qpt - params[j].qpt) / CAMath::Min(params[i].qpt, params[j].qpt)) < 0.15f;
```

Comment:
Currently, I check for 100cm difference in z (nominal z, t scaled linearly).
And then just some checks for the individual parameters. I assume I should better compute the covariance for the parameter set and do a full cov comparison and then cut on the chi2.
As is, this removes already ~2% of clusters at reasonable fake removes.
But for many of the short tracks the track fit is pretty poor, so the matching will not be so easy.